bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/profiles/TimestampOffsetProfile.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.thing.internal.profiles;
+
+import java.time.DateTimeException;
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.internal.i18n.I18nProviderImpl;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.thing.profiles.ProfileCallback;
+import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.thing.profiles.ProfileTypeUID;
+import org.openhab.core.thing.profiles.StateProfile;
+import org.openhab.core.thing.profiles.SystemProfiles;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Applies the given parameter "offset" to a {@link DateTimeType} state.
+ *
+ * Options for the "timezone" parameter are provided by the {@link I18nProviderImpl}.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class TimestampOffsetProfile implements StateProfile {
+
+    static final String OFFSET_PARAM = "offset";
+    static final String TIMEZONE_PARAM = "timezone";
+
+    private final Logger logger = LoggerFactory.getLogger(TimestampOffsetProfile.class);
+
+    private final ProfileCallback callback;
+
+    private final Duration offset;
+    private @Nullable ZoneId timeZone;
+
+    public TimestampOffsetProfile(ProfileCallback callback, ProfileContext context) {
+        this.callback = callback;
+
+        Object offsetParam = context.getConfiguration().get(OFFSET_PARAM);
+        logger.debug("Configuring profile with {} parameter '{}'", OFFSET_PARAM, offsetParam);
+        if (offsetParam instanceof Number) {
+            Number bd = (Number) offsetParam;
+            offset = Duration.ofSeconds(bd.longValue());
+        } else if (offsetParam instanceof String) {
+            String s = (String) offsetParam;
+            offset = Duration.ofSeconds(Long.valueOf(s));
+        } else {
+            logger.error(
+                    "Parameter '{}' is not of type String or Number. Please make sure it is one of both, e.g. 3 or \"-1.4\".",
```

Comment:
Only long values are supported:

```suggestion
                    "Parameter '{}' is not of type String or Number. Please make sure it is one of both, e.g. 3 or \"-1\".",
```